src/TFTDashboard.cpp

```
@@ -0,0 +1,102 @@
+void TFTDashboard::renderCell(LGFX_Sprite* sprite, int x, int y, int w, int h,
+                               const TrayData& tray, bool small) {
+    uint32_t bgColor = (static_cast<uint32_t>(tray.color[0]) << 16) |
+                       (static_cast<uint32_t>(tray.color[1]) << 8) |
+                        static_cast<uint32_t>(tray.color[2]);
+    sprite->fillRect(x, y, w, h, bgColor);
+
+    uint32_t textColor = contrastTextColor(tray.color[0], tray.color[1], tray.color[2]);
+    sprite->setTextColor(textColor);
+    sprite->setTextDatum(MC_DATUM);
+
+    char label[4];
```

Clang (14.0.6)

[warning] do not declare C-style arrays, use std::array<> instead

(modernize-avoid-c-arrays)